cob_cartesian_controller/src/cartesian_controller_utils.cpp

```diff
@@ -0,0 +1,161 @@
+/*!
+ *****************************************************************
+ * \file
+ *
+ * \note
+ *   Copyright (c) 2015 \n
+ *   Fraunhofer Institute for Manufacturing Engineering
+ *   and Automation (IPA) \n\n
+ *
+ *****************************************************************
+ *
+ * \note
+ *   Project name: care-o-bot
+ * \note
+ *   ROS stack name: cob_control
+ * \note
+ *   ROS package name: cob_cartesian_controller
+ *
+ * \author
+ *   Author: Christoph Mark, email: christoph.mark@ipa.fraunhofer.de / christoph.mark@gmail.com
+ *
+ * \date Date of creation: July, 2015
+ *
+ * \brief
+ *   ...
+ *
+ ****************************************************************/
+
+#include <cob_cartesian_controller/cartesian_controller_utils.h>
+
+
+geometry_msgs::Pose CartesianControllerUtils::getPose(std::string target_frame, std::string source_frame)
+{
+    geometry_msgs::Pose pose;
+    tf::StampedTransform stamped_transform;
+    
+    stamped_transform = getStampedTransform(target_frame, source_frame);
+    
+    tf::pointTFToMsg(stamped_transform.getOrigin(), pose.position);
+    tf::quaternionTFToMsg(stamped_transform.getRotation(), pose.orientation);
+
+    return pose;
+}
+
+tf::StampedTransform CartesianControllerUtils::getStampedTransform(std::string target_frame, std::string source_frame)
+{
+    tf::StampedTransform stamped_transform;
+    bool transform = false;
+    
+    do
+    {
+        try
+        {
+            ros::Time now = ros::Time::now();
+            tf_listener_.waitForTransform(target_frame, source_frame, now, ros::Duration(0.5));
```

Comment:
waitForTransform has a boolean return value -> transform = tf_listener_.waitForTransform(target_frame, source_frame, now, ros::Duration(0.5)); would be sufficient. 

Furthermore it is not necessary to catch a exception when the waitForTransform return value is checked before calling lookup!